Userland/Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -55,23 +55,39 @@ void ArrayBuffer::visit_edges(Cell::Visitor& visitor)
 
 // 25.1.2.1 AllocateArrayBuffer ( constructor, byteLength ), https://tc39.es/ecma262/#sec-allocatearraybuffer
 // 1.1.2 AllocateArrayBuffer ( constructor, byteLength [, maxByteLength ] ), https://tc39.es/proposal-resizablearraybuffer/#sec-allocatearraybuffer
-ThrowCompletionOr<ArrayBuffer*> allocate_array_buffer(GlobalObject& global_object, FunctionObject& constructor, size_t byte_length, Optional<size_t> max_byte_length)
+ThrowCompletionOr<ArrayBuffer*> allocate_array_buffer(GlobalObject& global_object, FunctionObject& constructor, size_t byte_length, Optional<size_t> maybe_max_byte_length)
 {
-    (void)max_byte_length;
-    // 1. Let obj be ? OrdinaryCreateFromConstructor(constructor, "%ArrayBuffer.prototype%", « [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] »).
+    // 1. Let slots be « [[ArrayBufferData]], [[ArrayBufferByteLength]], [[ArrayBufferDetachKey]] ».
+    // 2. If maxByteLength is present, append [[ArrayBufferMaxByteLength]] to slots.
+
+    // 3. Let obj be ? OrdinaryCreateFromConstructor(constructor, "%ArrayBuffer.prototype%", slots).
     auto* obj = TRY(ordinary_create_from_constructor<ArrayBuffer>(global_object, constructor, &GlobalObject::array_buffer_prototype, nullptr));
 
-    // 2. Let block be ? CreateByteDataBlock(byteLength).
+    // 4. Let block be ? CreateByteDataBlock(byteLength).
     auto block = ByteBuffer::create_zeroed(byte_length);
     if (block.is_error())
         return global_object.vm().throw_completion<RangeError>(global_object, ErrorType::NotEnoughMemoryToAllocate, byte_length);
 
-    // 3. Set obj.[[ArrayBufferData]] to block.
+    // 5. Set obj.[[ArrayBufferData]] to block.
     obj->set_buffer(block.release_value());
 
-    // 4. Set obj.[[ArrayBufferByteLength]] to byteLength.
+    // 6. Set obj.[[ArrayBufferByteLength]] to byteLength.
+
+    // 7. If maxByteLength is present, then
+    if (maybe_max_byte_length.has_value()) {
+        auto max_byte_length = maybe_max_byte_length.release_value();
```

Comment:
I don't think we need to have the `maybe_` variable, just keep the old name for the parameter and do `*max_byte_length` to get the value once we know it has one.